csrc/device_lower/lower2device.cpp

```diff
@@ -395,19 +395,20 @@ void GpuLower::analysis(Fusion* fusion) {
     // Enable validation in the DEBUG build mode
 #ifdef NDEBUG
     // Not DEBUG build
-    IdModel id_model(
+    id_model_ = std::make_unique<IdModel>(
         fusion_,
         /*build_graphs=*/true,
         /*allow_self_mapping=*/false,
         /*validate=*/false);
 #else
     // DEBUG build
-    IdModel id_model(
+    id_model_ = std::make_unique<IdModel>(
```

Comment:
Since we always build the IdModel now, would it make sense to initialize it in the `GpuLower` constructor instead of using a `unique_ptr`?

Reply:
Could you give more details? This `analysis` function is called from the constructor.

Reply:
I meant we could have `IdModel id_model_;` as a member instead of unique_ptr since it doesn't ever need to hold a `nullptr`. We would initialize it in the member initializer list
```c++

GpuLower::GpuLower(Fusion* fusion, const CompileParams& cparams = CompileParams())
    : passes_(/*...*/), cparams_(cparams), id_model_(fusion) {
  // ...
}
```
Is this not possible because we need to create a new copy of the fusion when we create `kernel_`?

Reply:
I think that's possible too. Maybe one reason to prefer the current style is that there are multiple analyses being done in `analysis` and there are some dependencies between them. Putting everything into the constructor initializer may make the code a little uglier? 

At least right now since we also want to make it possible to disable building an IdModel, I think the current style makes more sense.

Reply:
Thanks. I agree let's keep it as is.